infra/pySourceCredChart/js/lib/pySourceCredChart.js

```diff
@@ -0,0 +1,320 @@
+var widgets = require('@jupyter-widgets/base');
+require('../style/pySourceCredChart.css');
+
+import * as d3 from 'd3';
+
+/// set constants
+const MAX_SIZE_PIXELS = 200
+
+var sourceCredChartModel = widgets.DOMWidgetModel.extend({
+	defaults:_.extend(_.result(this, 'widgets.DOMWidgetModel.prototype.defaults'),{
+		_model_name: 'sourceCredChartModel',
+		_view_name: 'sourceCredChartView',
+		_model_module: 'pySourceCredChart',
+		_view_module: 'pySourceCredChart',
+	})
+});
+
+var sourceCredChartView = widgets.DOMWidgetView.extend({
+	
+	render: function(){
+		var that = this;
```

Comment:
style nit: personally, I prefer to use fat arrow functions `(arg) => { return result; }` rather than `function(arg) { return result; }` because fat arrows preserve `this` context and so tracking that is unnecessary. no change requested.